### PR TITLE
cloud: add release notes for 20231031

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

Add TiDB Cloud release notes for 20231031.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by hfxsd — APPROVED** [review]

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 1]
```diff
@@ -8,6 +8,16 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
 
 This page lists the release notes of [TiDB Cloud](https://www.pingcap.com/tidb-cloud/) in 2023.
 
+## October 31, 2023
+
+**General changes**
+
+- Upgrade the default TiDB version of new [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated) clusters from [v7.1.1](https://docs.pingcap.com/tidb/v7.1/release-7.1.1) to [v7.1.2](https://docs.pingcap.com/tidb/v7.1/release-7.1.2).
+
```
Comment (qiancai) [inline tidb-cloud/tidb-cloud-release-notes.md:16]: The TiDB Dedicated version in https://docs.pingcap.com/tidbcloud/tidb-cloud-faq#what-versions-of-tidb-are-supported-on-tidb-cloud also needs to be updated.

Comment (Oreoxmt) [reply tidb-cloud/tidb-cloud-release-notes.md:16]: https://github.com/pingcap/docs/pull/15221/commits/a66ec989a82050099309b9e406c387649344fdcc PTAL again

File: tidb-cloud/tidb-cloud-faq.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ No.
 
 ### What versions of TiDB are supported on TiDB Cloud?
 
-- Starting from July 25, 2023, the default TiDB version for new TiDB Dedicated clusters is v7.1.1.
+- Starting from October 31, 2023, the default TiDB version for new TiDB Dedicated clusters is v7.1.1.
```
Comment (qiancai) [inline tidb-cloud/tidb-cloud-faq.md:44]:
```suggestion
- Starting from October 31, 2023, the default TiDB version for new TiDB Dedicated clusters is v7.1.2.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**